core/Period/Factory.php

```diff
@@ -168,10 +168,10 @@ public static function makePeriodFromQueryParams($timezone, $period, $date)
         } else {
             if (!($date instanceof Date)) {
                 if (preg_match('/^(now|today|yesterday|yesterdaySameTime|last[ -]?(?:week|month|year))$/i', $date)) {
-                    $date = Date::factory($date, $timezone);
-                } else {
-                    $date = Date::factory($date);
+                    // calculate the date taking the current time into account
+                    $date = Date::factory(Date::factory($date)->setTime(date('H:i:s', Date::getNowTimestamp()))->getTimestamp(), $timezone)->getDatetime();
                 }
+                $date = Date::factory($date);
```

Comment:
also can we add a comment or something here saying if `$date` is a `Date` (which can be created above), the time of day is stripped (which is not something someone would know unless they knew Date did that weird thing).

Reply:
I've changed the code above so it returns a date instead of a datetime. That makes it unneeded to know if date would cut off the time or not.

Reply:
nvm, this was old behavior. I think you changed it actually.